openpype/hosts/maya/plugins/publish/extract_maya_scene_raw.py

```diff
@@ -1,13 +1,117 @@
 # -*- coding: utf-8 -*-
 """Extract data as Maya scene (raw)."""
 import os
+import contextlib
 
 from maya import cmds
 
 from openpype.hosts.maya.api.lib import maintained_selection
 from openpype.pipeline import AVALON_CONTAINER_ID, publish
 
 
+def offset_node(node, offset):
+    node_type = cmds.nodeType(node)
+    if node_type.startswith("animCurve"):
+        cmds.keyframe(node, edit=True, relative=True, timeChange=offset)
+
+    elif node_type == "imagePlane":
+        node_attr = node + ".frameOffset"
+        cmds.setAttr(node_attr, cmds.getAttr(node_attr) + (-offset))
+
+    elif node_type == "timeSliderBookmark":
+        for attr in ["timeRangeStart", "timeRangeStop"]:
+            node_attr = "{}.{}".format(node, attr)
+            cmds.setAttr(node_attr, cmds.getAttr(node_attr) + offset)
+
+
+@contextlib.contextmanager
+def offset_scene(offset):
+    # Exit early for easier stacking of context managers.
+    if offset is None:
+        yield
+        return
+
+    nodes = cmds.ls(type=("animCurve", "imagePlane", "timeSliderBookmark"))
```

Comment:
I believe `animCurve` is the base class for **all** curves, including keyframes for Set Driven Keys - which I'm pretty sure you don't want to offset.